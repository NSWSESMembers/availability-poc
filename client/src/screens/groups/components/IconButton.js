import React from 'react';
import { TouchableOpacity, Text } from 'react-native';
import PropTypes from 'prop-types';

import Icon from '../../../components/Icon';
import styles from './styles';


const IconButton = ({ onPress, icon, text }) => (

  <TouchableOpacity
    style={styles.iconRightHolder}
    onPress={onPress}
  >
    <Icon style={styles.icon} name={icon} size={25} />
    <Text style={styles.leaveGroupText}>{text}</Text>
  </TouchableOpacity>
);

IconButton.propTypes = {
  onPress: PropTypes.func.isRequired,
  icon: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
};

export default IconButton;
