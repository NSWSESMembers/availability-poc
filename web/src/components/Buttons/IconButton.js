import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import Button from '@material-ui/core/Button';
import Icon from '@material-ui/core/Icon';

import { withStyles } from '@material-ui/core/styles';

import styles from './IconButton.styles';

const IconButton = ({ classes, label, icon, onClick }) => (
  <Button variant="raised" size="small" color="primary" className={classes.button} onClick={onClick}>
    <Icon className={classNames(classes.leftIcon, classes.iconSmall)}>{icon}</Icon>
    Edit
  </Button>
);

IconButton.propTypes = {
  classes: PropTypes.shape({}).isRequired,
  label: PropTypes.string,
  icon: PropTypes.string,
  onClick: PropTypes.func.isRequired,
};

export default withStyles(styles)(IconButton);

