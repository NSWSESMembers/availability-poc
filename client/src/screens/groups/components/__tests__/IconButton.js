import React from 'react';
import renderer from 'react-test-renderer';

import IconButton from '../IconButton';

const noop = () => {};

test('renders correctly', () => {
  const tree = renderer.create(
    <IconButton
      onPress={noop}
      icon="fa-sign-out"
      text="text"
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
