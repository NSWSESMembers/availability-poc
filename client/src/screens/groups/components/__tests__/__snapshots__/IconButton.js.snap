// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "column",
      "justifyContent": "center",
      "opacity": 1,
      "paddingLeft": 8,
      "width": 50,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 25,
        },
        Object {
          "alignSelf": "center",
        },
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    ï‚‹
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignItems": "center",
        "color": "#6B6B6B",
        "flex": 0,
        "fontSize": 7,
        "includeFontPadding": false,
        "justifyContent": "center",
        "textAlign": "center",
      }
    }
  >
    text
  </Text>
</View>
`;
